samples/compiler_plugins/xnnpack/src/xnnpack/Conversion/StablehloToXnnpack.cpp

```diff
@@ -118,21 +118,25 @@ LogicalResult parsePatternFromFile(MLIRContext *context,
 }
 
 // If `val` is the result of a `stablehlo.convert`, get the unconverted value.
-static Value getUnconvertedValue(Value val) {
+static Value getUnconvertedValueOfElemType(Value val, Type elementType) {
+  if (val.getType().cast<RankedTensorType>().getElementType() == elementType)
+    return val;
   if (auto convert = val.getDefiningOp<mlir::stablehlo::ConvertOp>())
-    return convert.getOperand();
-  return val;
+    return getUnconvertedValueOfElemType(convert.getOperand(), elementType);
```

Comment:
Is there a real case with a series of convert ops?
Even if there is, we should be very careful in merging the convert ops. I'd suggest we don't do the recursion here since such a folding should be done by a separate canonicalization with a careful inspection.

Reply:
I did the recursion purely for stylistic reasons. It makes the code, IMO, less verbose and more readable. I haven't seen any cases at the moment of a series of convert ops.

> we should be very careful in merging the convert ops.

What do you mean by merging? I'm not sure I understand the concern.